File: Phase-2/frontend/src/types/task.ts
/**
 * Task types
 *
 * @spec specs/003-todo-frontend/spec.md (FR-015, FR-016, FR-017, FR-020, FR-024)
 */

export type TaskStatus = 'pending' | 'in-progress' | 'completed' | 'deleted';
export type TaskPriority = 'low' | 'medium' | 'high';

export interface Task {
  id: string;
  user_id: string;
  title: string;
  description: string | null;
  status: TaskStatus;
  priority: TaskPriority;
  created_at: string;
  updated_at: string;
}

export interface TaskListState {
  tasks: Task[];
  isLoading: boolean;
  error: string | null;
}

export interface TaskAnalytics {
  statusDistribution: {
    pending: number;
    'in-progress': number;
    completed: number;
  };
  priorityDistribution: {
    low: number;
    medium: number;
    high: number;
  };
  completionTrend: {
    date: string;
    completed: number;
  }[];
}

File: Phase-3/backend/app/models/task.py
"""
Task model — Phase 3 separate database (not shared with Phase 2).
@spec: specs/001-chatbot-backend/spec.md (FR-009)
"""
from datetime import datetime
from typing import Optional
from uuid import UUID, uuid4

from sqlmodel import Field, SQLModel


class Task(SQLModel, table=True):
    """
    Task table for Phase 3. Separate from Phase 2.
    status: 'pending' | 'in_progress' | 'completed' | 'deleted'
    """
    __tablename__ = "tasks"

    id: UUID = Field(default_factory=uuid4, primary_key=True, nullable=False)
    user_id: str = Field(nullable=False, index=True, max_length=255)
    title: str = Field(nullable=False, max_length=500)
    description: Optional[str] = Field(default=None, nullable=True)
    status: str = Field(default="pending", nullable=False, max_length=50, index=True)
    priority: Optional[str] = Field(default="medium", nullable=True, max_length=50)
    created_at: datetime = Field(default_factory=datetime.utcnow, nullable=False)
    updated_at: datetime = Field(default_factory=datetime.utcnow, nullable=False)
